backend-implicaction/src/main/java/com/dynonuggets/refonteimplicaction/service/WorkExperienceService.java

```diff
@@ -49,4 +51,15 @@
                 .map(workExperienceAdapter::toDtoWithoutUser)
                 .collect(toList());
     }
+
+    public void deleteExperience(Long idToDelete, Long currentUserId) {
+        WorkExperience workExperience = workExperienceRepository.findById(idToDelete)
+                .orElseThrow(() -> new NotFoundException("Aucune expérience avec l'Id : " + idToDelete + " trouvée."));
+
+        if (!workExperience.getUser().getId().equals(currentUserId)) {
+            throw new UnauthorizedException("Impossible de supprimer les expériences d'un autre utilisateur.");
+        } else {
+            workExperienceRepository.delete(workExperience);
+        }
```

Comment:
```suggestion

            workExperienceRepository.delete(workExperience);
```
pas besoin de mettre un else si tu throw une exception.